src/streaming.rs

```diff
@@ -0,0 +1,270 @@
+use ::open_vcdiff_sys;
+
+use std::error;
+use std::fmt;
+use std::marker::PhantomData;
+use std::mem;
+use std::os::raw::c_void;
+use std::slice;
+use std::result;
+
+#[derive(Debug,Copy,Clone,PartialEq,Eq)]
+pub struct Error(&'static str);
+
+impl error::Error for Error {
+    fn description(&self) -> &str {
+        self.0
+    }
+}
+
+impl fmt::Display for Error {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "vcdiff error: {}", self.0)
+    }
+}
+
+pub type Result<T> = result::Result<T,Error>;
+
+pub trait Output {
```

Comment:
I'm not sure under what circumstances it calls `size()` or `clear()` right now. If we don't need them, then sure.

Exception handling is a good point – right now, an `Output` that panics will unwind straight into (through?) C++. That's undefined, so `unsafe` callbacks need to `catch_unwind` before returning to C++… but what do they do with the panic? Store it `thread_local!` and re-panic once the call into C++ returns? This ties into error handling in general.